### PR TITLE
Clamp certdir cname and aha provisioning name values (SYN-6634)

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Ensure that we do not generate CSRs or make certificates with potentially invalid X509 Common Name length strings.
- Update AHA to prevent creating invalid provisioning requests.

### CODE REVIEW

File: synapse/lib/aha.py [hunk 1]
```diff
@@ -1039,6 +1051,10 @@ async def addAhaSvcProv(self, name, provinfo=None):
 
         hostname = f'{name}.{netw}'
 
+        if len(hostname) > 64:
+            mesg = f'Computed hostname value must not exceed 64 characters in length. {hostname=}, len={len(hostname)}'
```
Comment (invisig0th) [inline synapse/lib/aha.py:1055]:
```suggestion
            mesg = f'Hostname value must not exceed 64 characters in length. {hostname=}, len={len(hostname)}'
```

File: synapse/lib/aha.py [hunk 2]
```diff
@@ -1140,8 +1156,15 @@ async def addAhaUserEnroll(self, name, userinfo=None, again=False):
             mesg = 'AHA server requires aha:network configuration.'
             raise s_exc.NeedConfValu(mesg=mesg)
 
+        if not name:
+            raise s_exc.BadArg(mesg='Empty name values are not allowed for provisioning.')
+
         username = f'{name}@{ahanetw}'
 
+        if len(username) > 64:
+            mesg = f'Computed username value must not exceed 64 characters in length. username={username}, len={len(username)}'
```
Comment (invisig0th) [inline synapse/lib/aha.py:1165]:
```suggestion
            mesg = f'Username value must not exceed 64 characters in length. username={username}, len={len(username)}'
```

File: synapse/lib/certdir.py [hunk 3]
```diff
@@ -1501,6 +1501,11 @@ def _genPrivKey(self) -> c_rsa.RSAPrivateKey:
         return c_rsa.generate_private_key(65537, self.crypto_numbits)
 
     def _genCertBuilder(self, name: str, pubkey: c_types.PublicKeyTypes) -> c_x509.CertificateBuilder:
+
+        if not 1 <= len(name) <= 64:
+            mesg = f'certificate name values must be between 1-64 characters. got name={name}, len={len(name)}'
```
Comment (invisig0th) [inline synapse/lib/certdir.py:1506]:
```suggestion
            mesg = f'Certificate name values must be between 1-64 characters. got name={name}, len={len(name)}'
```

File: synapse/lib/certdir.py [hunk 4]
```diff
@@ -1515,6 +1520,10 @@ def _genCertBuilder(self, name: str, pubkey: c_types.PublicKeyTypes) -> c_x509.C
 
     def _genPkeyCsr(self, name: str, mode: str, outp: OutPutOrNone = None) -> bytes:
 
+        if not 1 <= len(name) <= 64:
+            mesg = f'csr name values must be between 1-64 characters. got name={name}, len={len(name)}'
```
Comment (invisig0th) [inline synapse/lib/certdir.py:1524]:
```suggestion
            mesg = f'CSR name values must be between 1-64 characters. got name={name}, len={len(name)}'
```

**Review by invisig0th — CHANGES_REQUESTED** [review]
doc nits

**Review by Cisphyx — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**